packages/calling/calling/src/CallingClient/CallingClient.ts

```diff
@@ -195,40 +89,73 @@ export class CallingClient extends Eventing<CallingClientEventTypes> implements
     if (!this.sdkConnector.getWebex()) {
       SDKConnector.setWebex(webex);
     }
-
+    this.mutex = new Mutex();
     this.webex = this.sdkConnector.getWebex();
 
     this.sdkConfig = config;
     const serviceData = this.sdkConfig?.serviceData?.indicator
       ? this.sdkConfig.serviceData
       : {indicator: ServiceIndicator.CALLING, domain: ''};
 
+    const logLevel = this.sdkConfig?.logger?.level ? this.sdkConfig.logger.level : LOGGER.ERROR;
     this.validateServiceData(serviceData);
 
-    this.registration = createRegistration(this.webex, serviceData);
+    this.registration = createRegistration(
+      this.webex,
+      serviceData,
+      this.mutex,
+      (event: EVENT_KEYS, deviceInfo?: IDeviceInfo, clientError?: CallingClientError) => {
```

Comment:
Could we maybe have a function defined for this in calling client and refer to the function here instead of putting the entire code in this line?